pennylane_lightning/src/gates/cpu_kernels/avx_common/ApplyControlledPhaseShift.hpp

```diff
@@ -0,0 +1,234 @@
+// Copyright 2022 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @file
+ * Defines ControlledPhaseShift gate
+ */
+#pragma once
+#include "AVXUtil.hpp"
+#include "BitUtil.hpp"
+#include "Permutation.hpp"
+#include "Util.hpp"
+
+#include <immintrin.h>
```

Comment:
Do we need the intrinsic header file here (and all above/below)? Should that not be encapsulated in the header files making the specific intrinsic calls, and be redundant here?

Reply:
Same thoughts